scipy/stats/_sobol.pyx

```diff
@@ -0,0 +1,351 @@
+from __future__ import division, absolute_import
+
+cimport cython
+cimport numpy as cnp
+
+import os
+import numpy as np
+
+# Parameters are linked to the direction numbers list.
+# See `initialize_direction_numbers` for more details.
+# Declared using DEF to be known at compilation time for ``poly`` et ``vinit``
+DEF MAXDIM = 21201  # max number of dimensions
+DEF MAXDEG = 18  # max polynomial degree
+DEF MAXBIT = 30  # max number of bits
+
+
+# Needed to be accessed with python
+cdef extern from *:
+    """
+    int MAXDIM_DEFINE = 21201;
+    int MAXDEG_DEFINE = 18;
+    int MAXBIT_DEFINE = 30;
+    """
+    int MAXDIM_DEFINE  # max number of dimensions
+    int MAXDEG_DEFINE  # max polynomial degree
+    int MAXBIT_DEFINE  # max number of bits
+
+_MAXDIM = MAXDIM_DEFINE
+_MAXDEG = MAXDEG_DEFINE
+_MAXBIT = MAXBIT_DEFINE
+
+cdef int poly[MAXDIM]
+cdef int vinit[MAXDIM][MAXDEG]
+
+cdef int LARGEST_NUMBER = 2 ** MAXBIT  # largest possible integer
+cdef float RECIPD = 1.0 / LARGEST_NUMBER  # normalization constant
+
+cdef bint is_initialized = False
+
+
+def initialize_direction_numbers():
+    """Load direction numbers.
+
+    Direction numbers obtained using the search criterion D(6)
+    up to the dimension 21201. This is the recommended choice by the authors.
+
+    Original data can be found at https://web.maths.unsw.edu.au/~fkuo/sobol/.
+    The C-code generated from putting the numbers in as literals is obscenely
+    large/inefficient. The data file was thus packaged and save as an .npz data
+    file for fast loading using the following code (this assumes that the file
+    https://web.maths.unsw.edu.au/~fkuo/sobol/new-joe-kuo-6.21201 is present in
+    the working directory):
+
+        import pandas as pd
+        import numpy as np
+
+        # read in file content
+        with open("./new-joe-kuo-6.21201", "r") as f:
+            lines = f.readlines()
+
+        rows = []
+
+        # parse data from file line by line
+        for l in lines[1:]:
+            nums = [int(n) for n in l.replace(" \n", "").split()]
+            d, s, a = nums[:3]
+            vs = {f"v{i}": int(v) for i,v in enumerate(nums[3:])}
+            rows.append({"d": d, "s": s, "a": a, **vs})
+
+
+        # read in as dataframe, explicitly use zero values
+        df = pd.DataFrame(rows).fillna(0).astype(int)
+
+        # peform conversion
+        df["poly"] = 2 ** df["a"] + 2** df["s"] + 1
```

Comment:
@tupui just realized there is a terrible typo here, this should be `2 * df["a"]` not `2 ** df["a"]`. Let me fix this and provide the right reference. I'll also make sure this didn't make it in the data file....

Reply:
Thank you for noticing this @balandat! If you push something, could you take care of the conflict? Thanks

Reply:
just pushed the typo update, what conflict? with master?

Turns out this is just a typo, the data file is correct (this would have caused a lot of issues)

Reply:
Good news! Yes I meant the `config.yml`

Reply:
ok let me fix it